### DISCUSSION TITLE
Is it possible to change the Energy?

### DISCUSSION THREAD

**Discussion by danymexi** [alengwenus/ha-sma-ev-charger#9] [Ideas]
Hello,

From the SMA ev web page I can change the energy the ev charger gives to the car. 

Is it possible to replicate? So I can set the power control system to increase/decrease the power of the ev charger based on the free energy grid I have.

thanks,
Daniele

**Comment by alengwenus** [comment]
The power limitation as well as the amount of energy are both implemented. The entity for power limitation is disabled and therefore hidden by default. It can be enabled in the integration/devices settings. The same for the current limitation.
The setting for the amount of energy should be visible by default.

**Reply by danymexi** [reply]
Hi,

I activated it but it is not changing the power or current.

Do you know if I’m doing something wrong? 
![IMG_6204](https://github.com/alengwenus/ha-sma-ev-charger/assets/116388889/c020f28e-6e20-4b8a-a660-dcbf27ac3469)

**Reply by alengwenus** [reply]
I wonder if setting the active power limit to 2000W is ok. I have a SMA EV Charger 22 and the minimal allowed power is 4500W. Trying to set a lower value does not work at all.
However, I tested it again with correct values and it works for me. Hint: The number field only sets the value if the field‘s focus is lost (e.g. click somewhere else in the UI).

**Reply by danymexi** [reply]
Hi alengwenus,

Question: I'm creating an automation to auto change the entity_id: number.sma_ev_charger_7_4_ac_current_limitation by +1 or -1 based on the current power from the grid.

I'm trying to use developer service:
service: input_number.increment
target:
  entity_id: number.sma_ev_charger_7_4_ac_current_limitation
data: {}

But it seems it doesn't work. Do you have any different ways to manage this?

Thanks a lot for suggestions :)
